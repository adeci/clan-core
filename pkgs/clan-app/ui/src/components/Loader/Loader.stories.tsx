import type { Meta, StoryObj } from "@kachurun/storybook-solid";
import { Loader, LoaderProps } from "@/src/components/Loader/Loader";

const meta: Meta<LoaderProps> = {
  title: "Components/Loader",
  component: Loader,
};

export default meta;

type Story = StoryObj<LoaderProps>;

export const Primary: Story = {
  args: {
    hierarchy: "primary",
  },
};

export const Secondary: Story = {
  args: {
    hierarchy: "secondary",
  },
};
