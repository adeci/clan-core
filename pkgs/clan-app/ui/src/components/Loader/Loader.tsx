import "./Loader.css";
import cx from "classnames";

export type Hierarchy = "primary" | "secondary";

export interface LoaderProps {
  hierarchy?: Hierarchy;
}

export const Loader = (props: LoaderProps) => {
  return (
    <div class={cx("loader", props.hierarchy || "primary")}>
      <div class="wrapper">
        <div class="parent"></div>
      </div>
      <div class="child"></div>
    </div>
  );
};
