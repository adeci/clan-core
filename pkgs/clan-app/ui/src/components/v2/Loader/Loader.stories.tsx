import type { Meta, StoryObj } from "@kachurun/storybook-solid";
import { Loader, LoaderProps } from "@/src/components/v2/Loader/Loader";

const meta: Meta<LoaderProps> = {
  title: "Components/Loader",
  component: Loader,
};

export default meta;

type Story = StoryObj<LoaderProps>;

export const Primary: Story = {
  args: {
    hierarchy: "primary",
  },
};

export const Secondary: Story = {
  args: {
    hierarchy: "secondary",
  },
};
