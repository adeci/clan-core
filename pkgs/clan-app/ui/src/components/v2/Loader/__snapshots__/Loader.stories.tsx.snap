// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Loader Primary smoke-test 1`] = `
<div class="loader primary">
  <div class="wrapper">
    <div class="parent">
    </div>
  </div>
  <div class="child">
  </div>
</div>
`;

exports[`Components/Loader Secondary smoke-test 1`] = `
<div class="loader secondary">
  <div class="wrapper">
    <div class="parent">
    </div>
  </div>
  <div class="child">
  </div>
</div>
`;
